src/Integration/LocalServices/SourceControlledFileSystem.cs

```diff
@@ -0,0 +1,175 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="SourceControlledFileSystem.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace SonarLint.VisualStudio.Integration
+{
+    internal class SourceControlledFileSystem : ISourceControlledFileSystem
```

Comment:
It would be worth adding some brief notes about the implementation e.g. that this class is used regardless of the type of source control system, or whether the project is under source control.